com.unity.render-pipelines.high-definition/Editor/Lighting/HDLightUI.cs

```diff
@@ -126,23 +138,52 @@ static HDLightUI()
                 )
             );
 
-            Type lightMappingType = typeof(Lightmapping);
-            var getLightingSettingsOrDefaultsFallbackInfo = lightMappingType.GetMethod("GetLightingSettingsOrDefaultsFallback", BindingFlags.Static | BindingFlags.NonPublic);
-            var getLightingSettingsOrDefaultsFallbackLambda = Expression.Lambda<Func<LightingSettings>>(Expression.Call(null, getLightingSettingsOrDefaultsFallbackInfo));
-            GetLightingSettingsOrDefaultsFallback = getLightingSettingsOrDefaultsFallbackLambda.Compile();
-
-            PresetInspector = CED.Group(
+            var presetInspector = CED.Group(
                 CED.Group((serialized, owner) =>
                     EditorGUILayout.HelpBox(s_Styles.unsupportedPresetPropertiesMessage, MessageType.Info)),
                 CED.Group((serialized, owner) => EditorGUILayout.Space()),
-                CED.FoldoutGroup(s_Styles.generalHeader, Expandable.General, k_ExpandedStatePreset, DrawGeneralContent),
-                CED.FoldoutGroup(s_Styles.emissionHeader, Expandable.Emission, k_ExpandedStatePreset, DrawEmissionContentForPreset),
+                CED.FoldoutGroup(s_Styles.generalHeader, Expandable.General, k_ExpandedStatePreset, CED.Group(
+                    (s, e) => DrawLightType(s, e, true),
+                    DrawGeneralContent)),
+                CED.FoldoutGroup(s_Styles.emissionHeader, Expandable.Emission, k_ExpandedStatePreset, CED.Group(DrawColor, DrawEmissionContent)),
                 CED.FoldoutGroup(s_Styles.shadowHeader, Expandable.Shadows, k_ExpandedStatePreset, DrawEnableShadowMapInternal)
             );
+
+            Inspector = CED.GroupWithPreset(normalInspector, presetInspector);
+
+            Type lightMappingType = typeof(Lightmapping);
+            var getLightingSettingsOrDefaultsFallbackInfo = lightMappingType.GetMethod("GetLightingSettingsOrDefaultsFallback", BindingFlags.Static | BindingFlags.NonPublic);
+            var getLightingSettingsOrDefaultsFallbackLambda = Expression.Lambda<Func<LightingSettings>>(Expression.Call(null, getLightingSettingsOrDefaultsFallbackInfo));
+            GetLightingSettingsOrDefaultsFallback = getLightingSettingsOrDefaultsFallbackLambda.Compile();
         }
 
-        static void DrawGeneralContent(SerializedHDLight serialized, Editor owner)
+        // This scope is here mainly to keep pointLightHDType isolated
+        public struct LightTypeEditionScope : IDisposable
+        {
+            EditorGUI.PropertyScope lightTypeScope;
+            EditorGUI.PropertyScope pointLightScope;
+
+            public LightTypeEditionScope(Rect rect, GUIContent label, SerializedHDLight serialized, bool isPreset)
+            {
+                lightTypeScope = new EditorGUI.PropertyScope(rect, label, serialized.settings.lightType);
```

Comment:
I noticed lightTypeScope and pointLightScope are in different order here vs the old code. Not sure if it matters at all.

Reply:
I changed just in case